daemon/container_operations.go

```diff
@@ -695,8 +698,14 @@ func buildEndpointDNSNames(ctr *container.Container, aliases []string) []string
 	return sliceutil.Dedup(dnsNames)
 }
 
-func (daemon *Daemon) connectToNetwork(cfg *config.Config, container *container.Container, idOrName string, endpointConfig *network.EndpointSettings, updateSettings bool) (retErr error) {
+func (daemon *Daemon) connectToNetwork(ctx context.Context, cfg *config.Config, container *container.Container, idOrName string, endpointConfig *network.EndpointSettings, updateSettings bool) (retErr error) {
 	start := time.Now()
+	ctx, span := otel.Tracer("").Start(ctx, "daemon.connectToNetwork", trace.WithAttributes(
+		attribute.String("daemon.container.ID", container.ID),
+		attribute.String("daemon.container.name", container.Name),
+		attribute.String("daemon.network.idOrName", idOrName)))
+	defer span.End()
```

Comment:
> Also curious if container.Name could have a `/ `prefix here (we strip that further down), and if we want that to be included, or if we should strip that here as well (they're here for legacy reasons, but can be confusing)

Honestly this `/` prefix has always been super confusing for me, ie. I don't get why we have it in the first place. So I have no opinion about whether it should be included here.

Reply:
@akerouanton I think the `/` prefix was because of legacy links; `/container` would be a container; `/container/foo` would be a link 🫠 

There were some PRs to strip them in some places, but we still had to account for those legacy links (but we should revisit that); for this use-case, I think it's fine to already strip the prefix as it's "greenfield" territory, so we don't have to take legacy things into account.